src/codons.py

```diff
@@ -60,15 +69,25 @@ def translate_codons(codons: list[str]) -> list[str]:
     True
 
     """
-    # FIXME: Implement the function
-    return []
 
-
-def translate_dna(dna: str) -> str:
+    amino_list = []
+    
+    for codon in codons_list:
+        if codon.upper() not in CODON_MAP.keys():
+            amino_list = None
```

Comment:
Here, you could also just return `None`. There is nothing more to do if we find that we have an unknown codon, so we might as well get out of the function. Breaking instead can lead to all kinds of trouble if we try to treat `amino_list` as a list after we have set it to `None`.